### ISSUE TITLE
Simplify how-to guides by extracting generalizable steps

### ISSUE THREAD

**Issue by ptgott** [gravitational/teleport#37819]
## Applies To

All how-to guides in the docs, which you can identify by the presence of a `##Prerequisites` H2. There are a lot of them:

```bash
grep -Rl "^## Prerequisites" docs/pages | wc -l
     178
```

### The problem

How-to guides in the Teleport documentation assume that the reader is using these guides as a kind of checklist, where each guide needs to be self contained and include all possible paths.

The problem is that there are often several ways to complete a setup. How-to guides that incorporate all of these paths (or even some of them) end up becoming intimidating or confusing. 

For example, in the guide to [protecting RDS withTeleport](https://goteleport.com/docs/database-access/guides/rds/) (as of [the day I wrote this issue](https://github.com/gravitational/teleport/blob/26d657d51e42c984c689e20ff3203217a20ef5d8/docs/pages/database-access/guides/rds.mdx#L149-L275)), we need to support deploying the Teleport Database Service on Linux servers and on Kubernetes, with both auto-discovery and a static configuration. The result is prohibitively cluttered.

Another issue with having to document every step in every guide is that it becomes difficult to add support in our how-to guides for new environments. In particular, we have had trouble [adding Kubernetes instructions](https://github.com/gravitational/teleport/issues/15043) to our how-to guides.

### Proposed solution

**Treat the how-to guide as _education_, not a _checklist_.**

Use how-to guides to get users familiar with a general procedure that they can then apply to other situations with the help of reference guides. Each how-to guide should follow a single path and avoid complex branching. We can restructure our docs to extract common procedures into their own guides, and order these guides into clear learning tracks for users to follow.

Particularly in agent setups, some procedures are generalizable across Teleport features:

- Configuring IAM for Teleport services running on EC2 instances.
- Configuring service account roles for Teleport agents on EKS.
- Deploying agents on Linux servers with a particular configuration (regardless of the configuration).
- Deploying `teleport-kube-agent` with a particular configuration (regardless of the configuration).
- Using static configuration vs. dynamic resources.

In other words, once a user understands the procedures for configuring permissions, deploying agents, etc., feature-specific how-to guides can be far simpler. We could even replace certain how-to guides with better reference docs. For example, we could organize reference documentation for agent service configurations by feature, explaining how each feature works and which configuration fields apply to it.

At the same time, generalizable steps like configuring IAM permissions do depend on the Teleport feature a user is setting up. A typical setup involves:

- Configuring a Teleport agent to enable a feature, which requires certain cloud API actions.
- Configuring permissions on the Teleport agent's cloud provider so the agent can access the API actions it needs.

In this case, we can structure the prerequisites of a guide to require following an _earlier_ guide in a learning track, e.g., setting up IAM, with the expectation that the user assign certain policies to a role.

One longstanding recommendation for how-to guides in the Teleport style guide is to prevent disrupting the user's focus by adding links to the body text of the guide. I think we can preserve this guidance—which I agree with—by requiring the user to follow other guides only in the Prerequisites sections of our how-to guides.

## How will we know this is resolved?

The deliverable for this issue is a more concrete plan that involves:
- An MVP
- Possible steps for expanding the MVP (i.e., other steps we can extract into generalizable steps)
- Plans to add any relevant items to the style guide and/or checks to our CI/CD workflows

Once that is complete, we can close this issue (but refer to it in subsequent planning docs for context).

As an example of an MVP, we could write two guides:

- How to configure AWS IAM for Teleport agents on Linux
- How to configure AWS IAM for Teleport agents on Kubernetes

We could then replace IAM instructions in how-to guides within one section of the docs (e.g., some "Database Access" guides) with Prerequisites items that link to the new guides.

## Related Issues

- https://github.com/gravitational/teleport/issues/15043
- https://github.com/gravitational/teleport/issues/24120
- https://github.com/gravitational/teleport/issues/15061

**Comment by ptgott** [comment]
I think it makes sense to close this in favor of implementing RFD 223, which will decouple agent deployment docs from agent configuration docs:
- https://github.com/gravitational/teleport/pull/59391
